logback/src/main/java/com/grafana/loki/BatchHandler.java

```diff
@@ -0,0 +1,159 @@
+package com.grafana.loki;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.concurrent.BlockingQueue;
+
+import javax.net.ssl.SSLContext;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.util.EntityUtils;
+
+import com.grafana.loki.batch.Batch;
+import com.grafana.loki.entry.Entry;
+import com.grafana.loki.http.HttpHandler;
+import com.grafana.loki.http.RetryHandler;
+
+import ch.qos.logback.core.Context;
+import ch.qos.logback.core.spi.ContextAwareBase;
+
+public class BatchHandler extends ContextAwareBase implements Runnable {
+	private volatile boolean running = true;
+
+	private BlockingQueue<Entry> entryQueue;
+
+	private BlockingQueue<Long> ticker;
+	
+	private LokiConfig config;
+
+	private Batch batch;
+
+	private String tenantID;
+
+	private URI url;
+
+	private String username;
+
+	private String password;
+
+	private long batchWait;
+
+	private int batchSizeBytes;
+
+	private SSLContext sslContext;
+
+	private Context ctx;
+
+	public BatchHandler(BlockingQueue<Entry> entryQueue, BlockingQueue<Long> ticker, LokiConfig config, Context ctx) {
+		this.entryQueue = entryQueue;
+		this.ticker = ticker;
+		this.config = config;
+		this.url = (URI) ctx.getObject("url");
+		this.tenantID = config.getTenantID();
+		this.username = config.getUsername();
+		this.password = config.getPassword();
+		this.batchWait = config.getBatchWaitSeconds() * 1000;
+		this.batchSizeBytes = config.getBatchSizeBytes();
+		this.sslContext = (SSLContext) ctx.getObject("sslContext");
+		this.ctx = ctx;
+		this.setContext(ctx);
+	}
+
+	@Override
+	public void run() {
+		try {
+			while (running) {
+				Entry e = entryQueue.poll();
+				Long ticker = this.ticker.poll();
```

Comment:
Hi, I built plugin jar from this pull request’s code and tired to use that with my application.
Then, I found that cpu utilization is always 100%.
I suggest that you use poll(long timeout, TimeUnit unit) instead of poll() or insert sleep in while loop.
What do you think about it ?